pkg/webhook/mutatingwebhook.go

```diff
@@ -0,0 +1,183 @@
+	// errorMsg is processed before MutatingWebhookConfiguration object is created.
+	errorMsg string
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+}
+
+// PullMutatingConfiguration pulls existing MutatingWebhookConfiguration from cluster.
+func PullMutatingConfiguration(apiClient *clients.Settings, name string) (*MutatingConfigurationBuilder, error) {
+	glog.V(100).Infof("Pulling existing MutatingWebhookConfiguration name %s from cluster", name)
+
+	builder := MutatingConfigurationBuilder{
+		apiClient: apiClient,
+		Definition: &admregv1.MutatingWebhookConfiguration{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MutatingWebhookConfiguration is empty")
+
+		builder.errorMsg = "MutatingWebhookConfiguration 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("MutatingWebhook object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given MutatingWebhookConfiguration object exists in the cluster.
+func (builder *MutatingConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns MutatingWebhookConfiguration object.
+func (builder *MutatingConfigurationBuilder) Get() (*admregv1.MutatingWebhookConfiguration, error) {
+	if valid, err := builder.validate(); !valid {
+		return &admregv1.MutatingWebhookConfiguration{}, err
+	}
+
+	mutatingWebhookConfiguration := &admregv1.MutatingWebhookConfiguration{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, mutatingWebhookConfiguration)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get MutatingConfigurationBuilder %s: %v", builder.Definition.Name, err)
+
+		return &admregv1.MutatingWebhookConfiguration{}, err
+	}
+
+	return mutatingWebhookConfiguration, err
+}
+
+// Delete removes a MutatingWebhookConfiguration from a cluster.
+func (builder *MutatingConfigurationBuilder) Delete() (*MutatingConfigurationBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the MutatingWebhookConfiguration %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("MutatingWebhookConfiguration cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete MutatingWebhookConfiguration: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// DeleteAndWait deletes an MutatingWebhookConfiguration and waits until it is removed from the cluster.
+func (builder *MutatingConfigurationBuilder) DeleteAndWait(timeout time.Duration) error {
```

Comment:
I don't think DeleteAndWait is relevant for this object. It removes immediately from cluster and doesn't have statuses like "Terminating" or "Pending".

Reply:
removed